contracts/vaults/base/RibbonVault.sol

```diff
@@ -251,17 +255,18 @@ contract RibbonVault is OptionsVaultStorage {
             "Insufficient balance"
         );
 
-        emit Deposit(msg.sender, amount, currentRound);
+        emit Deposit(creditor, amount, currentRound);
 
         Vault.DepositReceipt memory depositReceipt =
-            depositReceipts[msg.sender];
+            depositReceipts[creditor];
 
         // If we have an unprocessed pending deposit from the previous rounds, we have to process it.
         uint128 unredeemedShares =
             depositReceipt.getSharesFromReceipt(
                 currentRound,
                 roundPricePerShare[depositReceipt.round],
-                vaultParams.decimals
+                vaultParams.decimals,
+                vaultParams.initialSharePrice
```

Comment:
We can remove the `initialSharePrice` param entirely since deposits on the first round should not affect share price